### PR TITLE
Auto upgrade fixes

### PR DESCRIPTION
**Pull request by cevian**
Related to #329

### CODE REVIEW

File: pkg/pgmodel/extension.go [hunk 1]
```diff
@@ -234,10 +256,30 @@ func getSqlVersion(v semver.Version, extName string) string {
...
 // getNewExtensionVersion returns the highest version allowed by validRange
-func getNewExtensionVersion(availableVersions semver.Versions, validRange semver.Range) (semver.Version, bool) {
+func getNewExtensionVersion(extName string,
+	availableVersions semver.Versions,
+	defaultVersion semver.Version,
+	validRange semver.Range,
+	validCurrentVersion bool,
+	currentVersion semver.Version) (semver.Version, bool) {
 	//sort higher extensions first
 	sort.Sort(sort.Reverse(availableVersions))
```
Comment (Harkishen-Singh) [inline pkg/pgmodel/extension.go:266]: I don't see availableVersion is already sorted. So, shouldn't this be `sort.Sort(sort.Reverse(sort.SliceStabe(callback)))` ? Or a normal sort and then negative for loop.

File: pkg/pgmodel/extension.go [hunk 2]
```diff
@@ -234,11 +256,20 @@ func getSqlVersion(v semver.Version, extName string) string {
...
 	printedWarning := false
 	for i := range availableVersions {
+		/* skip any versions above the default version for auto-upgrade */
+		if availableVersions[i].GT(defaultVersion) {
```
Comment (Harkishen-Singh) [inline pkg/pgmodel/extension.go:270]: Could you please mention the significance of the `defaultVersion` here? The name default does not self explain the matter to me.

**Comment by Harkishen-Singh** [comment]
Also, should we have some unit-test here? Like, making a mock version that is greater in major and then trying to see if it is accepted or not by the function. WDYT?

File: pkg/pgmodel/extension.go (hunk 2 above)
Comment (cevian) [reply pkg/pgmodel/extension.go:270]: Default version is specified by the extension itself when installed into the system. It is usually the latest version but not always. It is installed by CREATE EXTENSION if no version is explicitly specified.

File: pkg/pgmodel/extension.go (hunk 1 above)
Comment (cevian) [reply pkg/pgmodel/extension.go:266]: Semver.Versions has Sort.Interface implemented. So this will sort in descending version order.

**Review by Harkishen-Singh — APPROVED** [review]
Lgtm!

**Review by JLockerman — APPROVED** [review]
LGTM

**Comment by JLockerman** [comment]
We might want to add some more explanatory comments at some point (such as what `defaultVersion` is) but those can happily wait to a followup PR

**Pull request merged**